### DISCUSSION TITLE
How to lazy load language files (i18n)?

### DISCUSSION THREAD

**Discussion by momander** [quasarframework/quasar#12253] [General - Components / Directives / etc] [answered]
Hi there!

I got vue-i18n to work with Quasar 2.0.0, Vue 3, the composition API, and <script setup>. I built a language switcher based on the **excellent** example at https://quasar.dev/options/app-internationalization#create-language-switcher. Thank you for providing that sample! All my i18n code is very similar to the examples in the Quasar docs, with minimal modifications.

My app supports 55 languages, each weighing in at 15-30 kB. The languages add a total of 800 kB to my "app.xxxxx.js" bundle when I build the application.

How do I lazy-load the language files to reduce bundle size? I figured out how to do it the old (pre-Quasar) version of my app by using Webpack imports and magic comments:

````await import(/* webpackChunkName: "locale-[request]" */ `./locales/${file}`)````

How would I lazy-load language files the Quasar way? It looks to me like `client/i18n/index.js` always imports all languages. Which files should I add code to, and what code?

Cheers

**Comment by yusufkandemir** [comment, marked as answer]
Hello 👋

As you've said, `src/i18n/index.js` and the code below make everything eagerly loaded.
```js
// src/boot/i18n.js
import messages from 'src/i18n' // <----

// ...
const i18n = createI18n({
  locale: 'en-US',
  messages // <----
})
```

By the way, the `src/boot/i18n.js` file and `src/i18n` folder are there just to get you started. You are free to use any structure you'd prefer or need.

So, in your case, instead of importing all of them at once, you can simply get rid of the current structure(_or just the `index.js` file_) and follow the [lazy loading guide on Vue I18n docs](https://vue-i18n.intlify.dev/guide/advanced/lazy.html). Hopefully, it should be clear enough considering you already are familiar with dynamic imports and magic comments.

**Comment by momander** [comment]
Thank you for a prompt answer!
